libyul/optimiser/LoadResolver.cpp

```diff
@@ -76,3 +105,62 @@ void LoadResolver::tryResolve(
 			if (inScope(*value))
 				_e = Identifier{locationOf(_e), *value};
 }
+
+void LoadResolver::tryEvaluateKeccak(
+	Expression& _e,
+	std::vector<Expression> const& _arguments
+)
+{
+	// Runtime saving: [PUSH32 32-byte-value] instead of [PUSH1 0 PUSH1 32 KECCAK256]
+	// Saves one KECCAK256 + one PUSH. Currently it is 30 + 6 + 3 = 39
+	unsigned runTimeGasSavings = GasCosts::keccak256Gas + GasCosts::keccak256WordGas + GasCosts::tier2Gas;
+	// Deploy time increase
+	// [PUSH32 32-byte-value]: 33 bytes
+	// instead of
+	// [PUSH1 0 PUSH1 32 KECCAK256]: 5 bytes
+	// Increase of 28 bytes.
+	// Current increase: 28 * 200: 56000
+	unsigned deployTimeGasLoss = 28 * GasCosts::createDataGas;
+	// We skip if the net savings are not positive
+	if (
+		m_expectedExecutionsPerDeployment.has_value() &&
+		bigint(*m_expectedExecutionsPerDeployment) * runTimeGasSavings < deployTimeGasLoss
```

Comment:
Why is `bigint` conversion needed?

Reply:
Just wondering if this could be made simpler by the following check

```
if (*m_expectedExecutionsPerDeployment < 435)
  return;
```

since gas saving/loss are const expressions.

Reply:
> Why is bigint conversion needed?

Was worried about overflow, since some contract writers seems to set very high `m_expectedExecutionsPerDeployment`.

> `if (*m_expectedExecutionsPerDeployment < 435)`

I did not want to hardcode the value since gas prices may change. Also, the RHS should be around 148.

Reply:
I suppose the `runs` parameter cannot exceed maximum of `size_t` datatype, right? That is the data type of the runs parameter in optimiser settings struct.

Reply:
> I suppose the runs parameter cannot exceed maximum of size_t datatype.

Yes. But when `runs` is `numeric_limits<size_t>::max()`, the multiplication would overflow or get into undefined behaviour. Doing it in `bigint` solves this issue. Alternatively, we could simply add an extra condition `size_t < 10000 && (....)`, if you think the `bigint` conversion is not nice.